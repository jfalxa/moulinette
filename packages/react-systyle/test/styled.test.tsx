import * as React from 'react'
import * as renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeContext } from '@emotion/core'

import Styled from '../src'

expect.addSnapshotSerializer(serializer)

// give a fixed classname to the component
Styled.className = 'styled'

it('creates an already usable systyle component', () => {
  const styled = renderer.create(<Styled>content</Styled>)
  const element = styled.toJSON()!

  expect(element.type).toBe('div')
  expect(element.props.className).toBe(Styled.toString().slice(1))
  expect(element.children).toEqual(['content'])
})

it('sets styling props', () => {
  const css = {
    id: 'styled',
    bg: 'red',
    px: 2,

    '&:hover': {
      bg: 'lime'
    }
  }

  const styled = renderer.create(<Styled {...css}>content</Styled>)

  expect(styled.toJSON()).toMatchSnapshot()
})

it('uses emotion theme context', () => {
  const theme = {
    colors: {
      test: 'red'
    }
  }

  const App = () => (
    <ThemeContext.Provider value={theme}>
      <Styled bg="test" />
    </ThemeContext.Provider>
  )

  const app = renderer.create(<App />)

  expect(app.toJSON()).toMatchSnapshot()
})
