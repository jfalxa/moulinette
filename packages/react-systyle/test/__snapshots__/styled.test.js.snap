// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets styling props 1`] = `
.emotion-0 {
  background-color: red;
  padding-left: 2px;
  padding-right: 2px;
}

.emotion-0:hover {
  background-color: lime;
}

<div
  className="suu7zrf782uf emotion-0"
  id="styled"
>
  content
</div>
`;

exports[`uses emotion theme context 1`] = `
.emotion-0 {
  background-color: red;
}

<div
  className="suu7zrf782uf emotion-0"
/>
`;
